jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/BadgeTests.java

```diff
@@ -1,47 +1,119 @@
+import static com.jdiai.tools.Timer.waitCondition;
+import static io.github.com.StaticSite.badgePage;
+import static io.github.com.enums.Colors.AMBER_ACCENT_2;
+import static io.github.com.enums.Colors.DEEP_PURPLE;
+import static io.github.com.enums.Colors.RED;
+import static io.github.com.pages.BadgePage.bellowPositionBadge;
+import static io.github.com.pages.BadgePage.buttonWithBadge;
+import static io.github.com.pages.BadgePage.customSizeBadge;
+import static io.github.com.pages.BadgePage.iconWithBadge;
+import static io.github.com.pages.BadgePage.largeBadge;
+import static io.github.com.pages.BadgePage.smallBadge;
+import static io.github.com.pages.BadgePage.textWithBadge;
+import static io.github.com.pages.BadgePage.toggleDisabledBadge;
+import static io.github.com.pages.BadgePage.toggleVisibilityBadge;
 
-// TODO Move to the new page
-@Ignore
 public class BadgeTests extends TestsInit {
 
-    @BeforeMethod(alwaysRun = true)
+    @BeforeMethod
     public void before() {
-        shouldBeLoggedIn();
-        angularPage.shouldBeOpened();
+        badgePage.open();
+        waitCondition(() -> badgePage.isOpened());
+        badgePage.checkOpened();
+        textWithBadge.show();
     }
 
     @Test
     public void basicBadgeTest() {
-        textWithBadge.show();
-        textWithBadge.badge().is().displayed();
-        textWithBadge.badge().has().text("4");
-        textWithBadge.has().color("Violet");
+        textWithBadge.is().displayed()
+                .has().text("4")
+                .has().color(DEEP_PURPLE.value());
     }
 
     @Test
     public void buttonBadgeTest() {
-        buttonWithBadge.show();
-        buttonWithBadge.badge().is().displayed();
-        buttonWithBadge.badge().has().text("8");
-        buttonWithBadge.has().color("Yellow");
+        buttonWithBadge.is().displayed()
+                .has().text("8")
+                .has().color(AMBER_ACCENT_2.value());
     }
 
     @Test
     public void iconBadgeTest() {
-        iconWithBadge.show();
-        iconWithBadge.badge().is().displayed();
-        iconWithBadge.badge().has().text("15");
-        iconWithBadge.has().color("Red");
+        iconWithBadge.is().displayed()
+                .has().text("15")
+                .has().color(RED.value());
+    }
+
+    @Test
+    public void badgeWithTextTest() {
+        customSizeBadge.is().displayed()
+                .has().text("Best")
+                .has().color(DEEP_PURPLE.value());
+
+    }
+
+    @Test
+    public void disabledBadgeTest() {
+        toggleDisabledBadge.is().disabled();
+
+        toggleDisabledBadge.click();
+        toggleDisabledBadge.is().enabled();
+
+        toggleDisabledBadge.click();
+        toggleDisabledBadge.is().disabled();
+    }
+
+    @Test
+    public void hiddenBadgeTest() {
+        toggleVisibilityBadge.is().displayed();
+        toggleVisibilityBadge.click();
+        toggleVisibilityBadge.is().hidden();
+        toggleVisibilityBadge.click();
```

Comment:
How it is possible to click on invisible element and be stable?

Reply:
The thing with the Badge element is that it addresses not the badge itself, but the element with the badge. So in this case it is the button with the id "toggle-visiblity-badge"
![image](https://github.com/jdi-testing/jdi-light/assets/16380954/22c8bc2d-a1da-4759-a5b3-f28214c258ca)

You said you disliked that in Badge class we get badge(). I discussed that with colleges, and they advised me to follow the patterns used in vuetify, but at least make badge() private.
I don`t know a better solution with the current strategy so far.